src/measure_theory/function/convergence_in_measure.lean

```diff
@@ -0,0 +1,364 @@
+/-
+Copyright (c) 2022 Rémy Degenne, Kexing Ying. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rémy Degenne, Kexing Ying
+-/
+
+import measure_theory.function.uniform_integrable
+
+/-!
+# Convergence in measure
+
+We define convergence in measure which is one of the many notions of convergence in probability.
+A sequence of functions `f` is said to converge in measure to some function `g`
+if for all `ε > 0`, the measure of the set `{x | ε ≤ dist (f i x) (g x)}` tends to 0 as `i`
+tends to infinity. Convergence in measure is most notably used in the formulation of the weak
+law of large numbers and is also useful in theorems such as the Vitali convergence theorem.
+This file provides some basic lemmas for working with convergence in measure and establishes
+some relations between convergence in measure and other notions of convergence.
+
+## Main definitions
+
+* `measure_theory.tendsto_in_measure (μ : measure α) (f : ι → α → E) (g : α → E)`: `f` converges
+  in `μ`-measure to `g`.
+
+## Main results
+
+* `measure_theory.tendsto_in_measure_of_tendsto_ae`: convergence almost everywhere in a finite
+  measure space implies convergence in measure.
+* `measure_theory.tendsto_in_measure.exists_seq_tendsto_ae`: if `f` is a sequence of functions
+  which converges in measure to `g`, then `f` has a subsequence which convergence almost
+  everywhere to `g`.
+* `measure_theory.tendsto_in_measure_of_tendsto_snorm`: convergence in Lp implies convergence
+  in measure.
+-/
+
+open topological_space filter
+open_locale nnreal ennreal measure_theory topological_space
+
+namespace measure_theory
+
+variables {α ι E : Type*} {m : measurable_space α} {μ : measure α}
+
+/-- A sequence of functions `f` is said to converge in measure to some function `g` if for all
+`ε > 0`, the measure of the set `{x | ε ≤ dist (f i x) (g x)}` tends to 0 as `i` tends to
+infinity. -/
+def tendsto_in_measure [has_dist E] {m : measurable_space α}
+  (μ : measure α) (f : ι → α → E) (l : filter ι) (g : α → E) : Prop :=
+∀ ε (hε : 0 < ε), tendsto (λ i, μ {x | ε ≤ dist (f i x) (g x)}) l (𝓝 0)
+
+lemma tendsto_in_measure_iff_norm [semi_normed_group E] {l : filter ι}
+  {f : ι → α → E} {g : α → E} :
+  tendsto_in_measure μ f l g
+  ↔ ∀ ε (hε : 0 < ε), tendsto (λ i, μ {x | ε ≤ ∥f i x - g x∥}) l (𝓝 0) :=
+by simp_rw [tendsto_in_measure, dist_eq_norm]
+
+namespace tendsto_in_measure
+
+variables [has_dist E] {l : filter ι} {f f' : ι → α → E} {g g' : α → E}
+
+protected lemma congr' (h_left : ∀ᶠ i in l, f i =ᵐ[μ] f' i) (h_right : g =ᵐ[μ] g')
+  (h_tendsto : tendsto_in_measure μ f l g) :
+  tendsto_in_measure μ f' l g' :=
+begin
+  intros ε hε,
+  suffices : (λ i, μ {x | ε ≤ dist (f' i x) (g' x)})
+      =ᶠ[l] (λ i, μ {x | ε ≤ dist (f i x) (g x)}),
+  { rw tendsto_congr' this,
+    exact h_tendsto ε hε, },
+  filter_upwards [h_left],
+  intros i h_ae_eq,
```

Comment:
can you merge master and use the new syntax for `filter_upwards`? Same thing below.